### PR TITLE
Fix : #234 Missing checks for address(0x0) when updating address stat…

### PR DESCRIPTION
**Pull request by ryzen-xp**
Summary of Changes Made
Input Validation in setL1L2CollectionMapping:
Added checks to ensure collectionL1 is not address(0) and collectionL2 is not 0 to prevent invalid mappings. Event Emission:
Emitted L1L2CollectionMappingUpdated after setting the mapping for transparency.

and  add unit test

### CODE REVIEW

File: apps/blockchain/ethereum/test/Bridge.t.sol [hunk 1]
```diff
@@ -600,5 +604,31 @@ contract BridgeTest is Test, IStarklaneEvent {
...
+   
+    function test_SetL1L2CollectionMapping() public {
+    
+        // Check for the event emission
+        vm.expectEmit(true, true, true, true);
+        emit L1L2CollectionMappingUpdated(collectionL1, snaddress.unwrap(collectionL2));
+       IStarklane(bridge).setL1L2CollectionMapping(collectionL1, collectionL2, false);
+    }
+
+    function testInvalidL1Address() public {
```
Comment (ptisserand) [inline apps/blockchain/ethereum/test/Bridge.t.sol:618]: Could you please change the test name to reflect it's checking `setL1L2CollectionMapping`

File: apps/blockchain/ethereum/test/Bridge.t.sol [hunk 2]
```diff
@@ -600,5 +604,31 @@ contract BridgeTest is Test, IStarklaneEvent {
...
+        vm.expectRevert("Invalid L1 address");
+       IStarklane(bridge).setL1L2CollectionMapping(address(0), collectionL2, false);
+    }
+
+    function testInvalidL2Address() public {
```
Comment (ptisserand) [inline apps/blockchain/ethereum/test/Bridge.t.sol:623]: Could you please change the test name to reflect it's checking `setL1L2CollectionMapping`

File: apps/blockchain/ethereum/test/Bridge.t.sol [hunk 3]
```diff
@@ -600,5 +604,31 @@ contract BridgeTest is Test, IStarklaneEvent {
...
+        vm.expectRevert("Invalid L2 address");
+        IStarklane(bridge).setL1L2CollectionMapping(collectionL1, Cairo.snaddressWrap(0), false);
+    }
+
+    function testInvalidUnwrappedL2Address() public {
```
Comment (ptisserand) [inline apps/blockchain/ethereum/test/Bridge.t.sol:628]: Could you please change the test name to reflect it's checking `setL1L2CollectionMapping`

**Review by ptisserand — CHANGES_REQUESTED** [review]
Thanks for your PR!
Only minor changes and it will good to be merged

**Comment by ryzen-xp** [comment]
Hello Sir @ptisserand  , 
            I change all  test name  , Can you recheck my PR request .     
        Thanks!

File: apps/blockchain/ethereum/src/Bridge.sol [hunk 4]
```diff
@@ -370,6 +370,8 @@ contract Starklane is IStarklaneEvent, UUPSOwnableProxied, StarklaneState, Stark
         snaddress collectionL2,
         bool force
     ) external onlyOwner {
+            require( collectionL1 != address(0x0), "Invalid L1 address");
```
Comment (ptisserand) [inline apps/blockchain/ethereum/src/Bridge.sol:373]: Sorry I miss this part in my previous review.
Could you please use solidity custom errors instead as other error:
```solidity
error BridgeNotEnabledError();
// ....

        if (!_enabled) {
            revert BridgeNotEnabledError();
        }
```

**Review by ptisserand — CHANGES_REQUESTED** [review]
Please use custom error instead of require

**Comment by ryzen-xp** [comment]
Hello Sir @ptisserand  ,
             I change  all  you want to change ,  Plz  Check my PR request .

File: apps/blockchain/ethereum/test/Bridge.t.sol [hunk 5]
```diff
@@ -600,5 +604,31 @@ contract BridgeTest is Test, IStarklaneEvent {
...
+    function test_SetL1L2CollectionMapping() public {
+    
+        // Check for the event emission
+        vm.expectEmit(true, true, true, true);
+        emit L1L2CollectionMappingUpdated(collectionL1, snaddress.unwrap(collectionL2));
+       IStarklane(bridge).setL1L2CollectionMapping(collectionL1, collectionL2, false);
+    }
+
+    function test_SetL1L2CollectionMappingWith_InvalidL1Address() public {
+         vm.expectRevert(abi.encodeWithSignature("InvalidL1AddressError()"));
```
Comment (ptisserand) [inline apps/blockchain/ethereum/test/Bridge.t.sol:619]: To match other tests, use:
```solidity
vm.expectRevert(InvalidL1AddressError.selector);
```

File: apps/blockchain/ethereum/test/Bridge.t.sol [hunk 6]
```diff
@@ -600,5 +604,31 @@ contract BridgeTest is Test, IStarklaneEvent {
...
+       IStarklane(bridge).setL1L2CollectionMapping(address(0), collectionL2, false);
+    }
+
+    function test_SetL1L2CollectionMappingWith_InvalidL2Address() public {
+        vm.expectRevert(abi.encodeWithSignature("InvalidL2AddressError()"));
```
Comment (ptisserand) [inline apps/blockchain/ethereum/test/Bridge.t.sol:624]: To match other tests, use:
```solidity
vm.expectRevert(InvalidL2AddressError.selector);
```

**Review by ptisserand — CHANGES_REQUESTED** [review]

**Comment by ryzen-xp** [comment]
@ptisserand  sir ,    can you  check again my PR plz.

**Review by ptisserand — APPROVED** [review]
Thanks for your contribution!

**Comment by ryzen-xp** [comment]
Thank you Sir

**Pull request merged**